src/mono/wasm/runtime/invoke-js.ts

```diff
@@ -84,9 +84,11 @@ export function mono_wasm_bind_js_function(function_name: MonoStringRef, module_
         (<any>bound_fn)[imported_js_function_symbol] = true;
         const fn_handle = fn_wrapper_by_fn_handle.length;
         fn_wrapper_by_fn_handle.push(bound_fn);
-        setI32(function_js_handle, <any>fn_handle);
+        setI32_unchecked(function_js_handle, <any>fn_handle);
+        wrap_no_error_root(is_exception, resultRoot);
```

Comment:
Why are you sometimes passing `resultRoot` to `wrap_no_error_root` and sometimes you don't?
What's the discriminator?

Reply:
In cases when the logical return type of the funtion is `void` the `result` is only used for passing exception. In case the method is not `void` and doesn't fail, the `result` should not be zeroed as it contains the return value.